test/integration/running/test_running.py

```diff
@@ -139,3 +140,91 @@ def test_logrotate(tt_cmd, tmpdir):
     # Check that the process was terminated correctly.
     instance_process_rc = instance_process.wait(1)
     assert instance_process_rc == 0
+
+
+def test_running_reread_config(tt_cmd, tmpdir):
+    # Copy the test application to the "run" directory.
+    test_app_path = os.path.join(os.path.dirname(__file__), "test_app", "test_app.lua")
+    shutil.copy(test_app_path, tmpdir)
+    inst_name = "test_app"
+    configPath = os.path.join(tmpdir, "tarantool.yaml")
+
+    # Create test config with restart_on_failure true.
+    with open(configPath, 'w') as f:
+        f.write('tt:\n  app:\n    restart_on_failure: true\n    log_dir: '+str(tmpdir)+"\n    log_maxsize: 10    \n    log_maxage: 1\n")
+
+    # Start an instance.
+    start_cmd = [tt_cmd, "start", inst_name, "--cfg", configPath]
+    instance_process = subprocess.Popen(
+        start_cmd,
+        cwd=tmpdir,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        text=True
+    )
+    start_output = instance_process.stdout.readline()
+    assert re.search(r"Starting an instance", start_output)
+
+    # Get pid of instance.
+    status_cmd = [tt_cmd, "status", inst_name]
+    status_rc, status_out = run_command_and_get_output(status_cmd, cwd=tmpdir)
+    assert status_rc == 0
+    assert re.search(r"RUNNING. PID: \d+.", status_out)
+
+    res = [int(i) for i in status_out if i.isdigit()]
+    pid = ""
+    for digit in res:
+        pid += str(digit)
+    print("Pid: ", pid)
+
+    # Wait for child process of instance to start.
+    logPath = os.path.join(tmpdir, inst_name + ".log")
+    flag = 0
+    while flag == 0:
+        try:
+            with open(logPath, "r") as f:
+                log = f.read()
+                if "started" in log:
+                    flag = 1
+        except:
+            continue
+    
+    # Kill instance child process.
+    kill_child_process(pid, tmpdir)
+
+    # Check status, it should be running, since instance restarts after failure.
+    status_cmd = [tt_cmd, "status", inst_name]
+    status_rc, status_out = run_command_and_get_output(status_cmd, cwd=tmpdir)
+    assert status_rc == 0
+    assert re.search(r"RUNNING. PID: \d+.", status_out)
+
+    # Create test config with restart_on_failure false.
+    with open(configPath, 'w') as f:
+        f.write("tt:\n  app:\n    restart_on_failure: false\n    log_dir: "+"str(tmpdir)+\n    log_maxsize: 10    \n    log_maxage: 1\n")
+    
+    # Wait for child process of instance to start.
```

Comment:
You already write this twice. Move this functionality to a separate function.

Reply:
moved